### PR TITLE
[DOCS-3085] Update logs permissions docs for Jan 18 release

### PR DESCRIPTION
**Pull request by urseberry**
### What does this PR do?
Update the 3 Logs pages that will be made out of date by the January 18 AAA code release. Follows the tracking spreadsheet/update plan here: https://docs.google.com/spreadsheets/d/1qAFjGVoxZusiOP8Qlevj5mCvELGuU4BYYc0OiZWLfAA/edit#gid=1935103315.

### Motivation
Docs need to be brought up to date to be consistent with January 18 AAA code release.

The January 18 changes are:
- removal of the `logs_public_config_api` permission. Instead, logs API permissions are governed by logs_generate_metrics, logs_modify_indexes, logs_write_archives, logs_write_pipelines, and user_access_manage
- There is no longer a permission called 'admin'. Permissions that used to be enabled by 'admin' are delegated to more specific roles and permissions.

### Preview
https://docs-staging.datadoghq.com/uchen/admin-release-1/logs/guide/access-your-log-data-programmatically/
https://docs-staging.datadoghq.com/uchen/admin-release-1/logs/guide/build-custom-reports-using-log-analytics-api/?tab=table
https://docs-staging.datadoghq.com/uchen/admin-release-1/logs/guide/logs-rbac-permissions/?tab=ui

### Additional Notes

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

File: content/en/logs/guide/logs-rbac-permissions.md [hunk 1]
```diff
@@ -197,26 +195,13 @@ In the example above:
...
-When assigning `team:audit` tags on all logs rehydrated from the `Audit Archive`, make sure that `Audit` role members who are restricted to read `team:audit`logs  can only access rehydrated content. For more details on how to add tags and rehydration, see the [Log Archive Setup section][12].
+When assigning `team:audit` tags on all logs rehydrated from the `Audit Archive`, make sure that `Audit` role members who are restricted to read `team:audit`logs can only access rehydrated content. For more details on how to add tags and rehydration, see the [Log Archive Setup section][12].
 
 For `service:ci-cd` logs that are rehydrated from the `Prod Archive`, note the following:
 
 * If you **do not** use the [Log Read Index Data](#logs-read-index-data) legacy permission, these logs are accessible for `CI-CD` role members.
 * If you **do** use the [Log Read Index Data](#logs-read-index-data) legacy permission, these logs are not accessible for `CI-CD` role members, as the resulting historical view is restricted to `PROD` and `ADMIN` role members.
 
-
-### `logs_public_config_api`
```
Comment (srosenthal-dd) [inline content/en/logs/guide/logs-rbac-permissions.md:208]: Wonder if we should add a temporary blurb about how this permission used to exist, but has been replaced by the underlying permissions?

File: content/en/logs/guide/logs-rbac-permissions.md [hunk 2]
```diff
@@ -17,9 +17,9 @@ Once you've created [RBAC roles for logs][1], assign or remove [permissions][2]
 {{< tabs >}}
 {{% tab "UI" %}}
 
-Assign or remove permission to a role directly by [updating the role in the Datadog application][1].
+Assign or remove permission to a role directly by [updating the role on the Datadog site][1].
 
-{{< img src="account_management/rbac/logs_permissions.png" alt="Logs Permissions"  style="width:75%;" >}}
+{{< img src="account_management/rbac/logs_permissions.png" alt="Logs Permissions" style="width:75%;" >}}
```
Comment (srosenthal-dd) [inline content/en/logs/guide/logs-rbac-permissions.md:22]: This screenshot is (or will be) out of date. But we don't have a final screenshot available yet. Consider removing it?

**Review by srosenthal-dd — APPROVED** [review]
No major concerns! Some comments inline but nothing serious.

I am planning on running the DB migration today, and if that goes well will change the permission checks to no longer uses the "logs_public_config_api" permission later this week.

File: content/en/logs/guide/access-your-log-data-programmatically.md [hunk 3]
```diff
@@ -28,11 +28,11 @@ The following examples are covered in this guide:
...
+- Use of the Logs Search API requires an [API key][2] and an [application key][3]. The user who created the application key must have the appropriate permission to access the data. To use the examples below, replace `<DATADOG_API_KEY>` and `<DATADOG_APP_KEY>` with your Datadog API key and your Datadog application key, respectively.
 
-- This guide features `curl` examples. Install [curl][3] if you do not have it installed, or reference additional language examples for this API endpoint in the [API documentation][1].
+- This guide features `curl` examples. Install [curl][4] if you do not have it installed, or reference additional language examples for this API endpoint in the [Logs API][1].
 
-**Note:** By default, all users with Admin, Standard, and Read-Only roles have access to all log data. But the search results might be restricted based on the permissions of the user owning the `<DATADOG_APP_KEY>`.
+**Note:** If you are in the Datadog EU site, use https://api.datadoghq.eu/api/  as the endpoint.
```
[the remark this answers is not in the document]
Comment (urseberry) [reply content/en/logs/guide/access-your-log-data-programmatically.md:35]: I noticed that the examples in this doc have code to automatically use the correct endpoint depending on which site is selected, so I'm going to remove this part entirely.

File: content/en/logs/guide/logs-rbac-permissions.md [hunk 4]
```diff
@@ -202,6 +201,17 @@ For `service:ci-cd` logs that are rehydrated from the `Prod Archive`, note the f
...
+### `logs_public_config_api`
+
+Datadog has deprecated the `logs_public_config_api` permission. 
+
+Five separate permissions control the ability to view, create, or modify log configuration through the Datadog API:
+* `logs_generate_metrics`
+* `logs_modify_indexes`
+* `logs_write_archives`
+* `logs_write_pipelines`
+* `user_access_manage`
```
Comment (srosenthal-dd) [inline content/en/logs/guide/logs-rbac-permissions.md:213]: Wonder if these should be links? Otherwise this looks great

**Review by srosenthal-dd — APPROVED** [review]
cool!

BTW the Datadog app has been updated so logs_public_config_api is gone/hidden, so you're free to merge this whenever you're ready as far as I'm concerned 👍

**Comment by urseberry** [comment]
This PR is ready for docs team review.

**Review by apigirl — APPROVED** [review]

**Pull request merged**